src/ripple/app/misc/AMMHelpers.h

```diff
@@ -145,12 +149,184 @@ withinRelativeDistance(Amt const& calc, Amt const& req, Number const& dist)
 }
 // clang-format on
 
-/** Finds takerPays (i) and takerGets (o) such that given pool composition
- * poolGets(I) and poolPays(O): (O - o) / (I + i) = quality.
- * Where takerGets is calculated as the swapAssetIn (see below).
- * The above equation produces the quadratic equation:
- * i^2*(1-fee) + i*I*(2-fee) + I^2 - I*O/quality,
- * which is solved for i, and o is found with swapAssetIn().
+namespace {
```

Comment:
Another choice that's been used elsewhere in the code base would be to use `namespace detail`, rather than the anonymous `namespace`.  I don't feel strongly about it, just presenting an option.

Reply:
I feel pretty strongly against anonymous namespaces in header files, because of internal linkage. Fine if everything inside such a namespace is explicitly `static` but otherwise can lead to ODR.